### PR TITLE
MaryiaF/68794/fix: Trackjs error: There was an error loading Google API script. - …

### PR DESCRIPTION
**Pull request by maryia-matskevich-deriv**
…Error: {"isTrusted":true}

## Changes

-   fix: Trackjs error: There was an error loading Google API script. - Error: {"isTrusted":true}

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/botPage/view/deriv/layout/Main/main.jsx [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
+import SidebarToggle from '../../components/SidebarToggle';
+import { updateActiveAccount, updateActiveToken, updateIsLogged } from '../../store/client-slice';
+import { setShouldReloadWorkspace, updateShowTour } from '../../store/ui-slice';
+import { getRelatedDeriveOrigin, isLoggedIn } from '../../utils';
 import BotUnavailableMessage from '../Error/bot-unavailable-message-page.jsx';
-import api from '../../api';
-import Helmet from 'react-helmet';
-import { getLanguage } from '../../../../../common/lang';
-import { translate } from '../../../../../common/utils/tools';
+import ToolBox from '../ToolBox';
```
Comment (maryiafrantsava-binary) [inline src/botPage/view/deriv/layout/Main/main.jsx:28]: to fix eslint error: error  Absolute imports should come before relative imports (fixed with option+shift+O)

File: src/botPage/view/deriv/layout/Main/main.jsx [hunk 2]
```diff
@@ -115,9 +114,10 @@ const Main = () => {
             $('.show-on-load').show();
             $('.barspinner').hide();
             window.dispatchEvent(new Event('resize'));
-            TrackJS.configure({
-                userId: document.getElementById('active-account-name')?.value,
-            });
+            const userId = document.getElementById('active-account-name')?.value;
+            if (userId) {
+                TrackJS.configure({ userId });
+            }
```
Comment (maryiafrantsava-binary) [inline src/botPage/view/deriv/layout/Main/main.jsx:120]: to fix warning:  [TrackJS] config.userId: property must be type string.

File: src/common/utils/tools.js [hunk 3]
```diff
@@ -137,6 +137,7 @@ export const loadExternalScript = (src, async = true, defer = true) =>
         script.src = src;
         script.async = async;
         script.defer = defer;
+        script.crossorigin = 'anonymous';
```
Comment (maryiafrantsava-binary) [inline src/common/utils/tools.js:140]: The current TrackJS error is **Script Error** type and it comes from init() google API here:

Screenshot 2022-12-08 at 18 02 11

To get more info about third side error from google API source I've added attribute **crossorigin**
* To check more info about this **Script Error** type read this article:
https://blog.sentry.io/2016/05/17/what-is-script-error/#the-cause-cross-origin-scripts
* To check that it is Script Error type you can check this article:
https://stackoverflow.com/questions/44815172/log-shows-error-object-istrustedtrue-instead-of-actual-error-data

**Review by sandeep-deriv — APPROVED** [review]

File: src/botPage/view/deriv/blockly-worksace.js [hunk 4]
```diff
@@ -277,8 +277,8 @@ const addBindings = blockly => {
             return;
         }
 
-        const token = document.getElementById('active-token').value;
-        const tokenObj = getToken(token);
+        const token = document.getElementById('active-token')?.value;
+        const tokenObj = token ? getToken(token) : false;
```
Comment (maryiafrantsava-binary) [inline src/botPage/view/deriv/blockly-worksace.js:281]: To fix TrackJS error Cannot read properties of null (reading 'value')

**Review by sandeep-deriv — APPROVED** [review]

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**